pkg/ansible/rloc.go

```diff
@@ -0,0 +1,37 @@
+package ansible
+
+import (
+	"bufio"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/MaibornWolff/iac-count/pkg/metrics"
+)
+
+type RlocCalculator struct {
+}
+
+func (calculator RlocCalculator) isFileValidForMetric(path string) bool {
+	return filepath.Ext(path) == ".yml"
+}
+
+func (calculator RlocCalculator) Analyze(path, content string) metrics.Metric {
+	if !calculator.isFileValidForMetric(path) {
+		return nil
+	}
+
+	re := regexp.MustCompile(`^\s*[^#].*`)
```

Comment:
Everything that is not an empty line or starts with a`#` is matched with this pattern.

Let's split analyze:
- `\s*` any amount of white spaces
- `[^#]` one character that is not a `#`
- `.*` any character in any amount

So the minimal string that matches is ` `, because it just has to satisfy the non-# part.

![grafik](https://user-images.githubusercontent.com/414431/77184339-f63a1e80-6acf-11ea-9947-f98a5d37995c.png)

The only line that doesn't match is line 7:

```
$ grep '^\s*[^#].*' re-examples
 
 #

 # 
                # match match
        # abc
```